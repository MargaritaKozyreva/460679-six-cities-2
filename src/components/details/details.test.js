import React from "react";
import Details from "./details.jsx";
import renderer from "react-test-renderer";

describe(`initial Details`, () => {
  const onTitleClickHandler = jest.fn();
  const mockProps = {
    details: {
      coordinates: [52.3909553943508, 4.85309666406198],
      id: 1,
      type: `Apartment`,
      title: `Canal View Prinsengracht[TEST-1]`,
      link: `offer/`,
      price: 140,
      mainImg: `img/apartment-01.jpg`,
      gallery: [
        `img/apartment-01.jpg`,
        `img/apartment-02.jpg`,
        `img/apartment-03.jpg`
      ],
      img: `img/apartment-01.jpg`,
      raiting: 3,
      roomClass: `Premium`,
      insideList: [
        `Wi-Fi`,
        `Washing machine`,
        `Towels`,
        `Heating`,
        `Coffee machine`,
        `Baby seat`,
        `Kitchen, Dishwasher`,
        `Cabel TV`,
        `Fridge`
      ],
      host: {
        hostImg: `img/avatar-angelina.jpg`,
        userName: `Angelina`,
        userStatus: `Pro`,
        description: [
          `A quiet cozy and picturesque that hides behind a a river by
            the unique lightness of Amsterdam. The building is green and
            from 18th century.`,
          `
            An independent House, strategically located between Rembrand
            Square and National Opera, but where the bustle of the city
            comes to rest in this alley flowery and colorful.
            `
        ]
      },
      reviews: [
        {
          id: 1,
          userName: `Max`,
          userImg: `img/avatar-max.jpg`,
          userRaiting: 4.5,
          text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
          createDate: `2019-05-21`
        },
        {
          id: 2,
          userName: `Maria`,
          userImg: `img/avatar-max.jpg`,
          userRaiting: 4,
          text: `Nice.`,
          createDate: `2019-04-24`
        }
      ]
    },
    cards: [
      {
        coordinates: [52.3909553943508, 4.85309666406198],
        id: 1,
        title: `Canal View Prinsengracht`,
        price: 140,
        mainImg: `img/apartment-01.jpg`
      }
    ],
    coords: [[1, 2]],
    onTitleClickHandler
  };

  it(`Details renders correctly`, () => {
    const tree = renderer
      .create(<Details {...mockProps} />, {
        createNodeMock: () => {
          let mapMock = document.createElement(`section`);
          mapMock.setAttribute(`id`, `map`);
          document.body.appendChild(mapMock);

          return mapMock;
        }
      })
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
