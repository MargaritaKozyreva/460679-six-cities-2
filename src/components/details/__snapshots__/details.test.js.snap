// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial Details Details renders correctly 1`] = `
<div
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-02.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="img/apartment-03.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Canal View Prinsengracht[TEST-1]
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height={33}
              width={31}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "96%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            3
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Entire place
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            3 Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 4 adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            140
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            Whats inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Wi-Fi
            </li>
            <li
              className="property__inside-item"
            >
              Washing machine
            </li>
            <li
              className="property__inside-item"
            >
              Towels
            </li>
            <li
              className="property__inside-item"
            >
              Heating
            </li>
            <li
              className="property__inside-item"
            >
              Coffee machine
            </li>
            <li
              className="property__inside-item"
            >
              Baby seat
            </li>
            <li
              className="property__inside-item"
            >
              Kitchen, Dishwasher
            </li>
            <li
              className="property__inside-item"
            >
              Cabel TV
            </li>
            <li
              className="property__inside-item"
            >
              Fridge
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height={74}
                src="img/avatar-angelina.jpg"
                width={74}
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
            <span
              className="property__user-status"
            >
              Pro
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              A quiet cozy and picturesque that hides behind a a river by
            the unique lightness of Amsterdam. The building is green and
            from 18th century.
            </p>
            <p
              className="property__text"
            >
              
            An independent House, strategically located between Rembrand
            Square and National Opera, but where the bustle of the city
            comes to rest in this alley flowery and colorful.
            
            </p>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                2
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="img/avatar-max.jpg"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "94%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        4.5
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2019-05-21"
                  >
                    2019-05-21
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="img/avatar-max.jpg"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Maria
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "94%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        4
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Nice.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2019-04-24"
                  >
                    2019-04-24
                  </time>
                </div>
              </li>
            </ul>
          </section>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "height": "579px",
          "width": "100%",
        }
      }
    >
      <section
        className="cities__map map"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      />
    </section>
  </div>
</div>
`;
